src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -2045,12 +2084,34 @@ JNI_METHOD(void, invoke)
 
     tlvBytesObj = static_cast<jbyteArray>(env->CallObjectMethod(invokeElement, getTlvByteArrayMethod));
     VerifyOrExit(!env->ExceptionCheck(), err = CHIP_JNI_ERROR_EXCEPTION_THROWN);
-    VerifyOrExit(tlvBytesObj != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
+    if (tlvBytesObj != nullptr)
+    {
+        jbyte * tlvBytesObjBytes = env->GetByteArrayElements(tlvBytesObj, nullptr);
+        VerifyOrExit(!env->ExceptionCheck(), err = CHIP_JNI_ERROR_EXCEPTION_THROWN);
+        length = static_cast<size_t>(env->GetArrayLength(tlvBytesObj));
+        VerifyOrExit(!env->ExceptionCheck(), err = CHIP_JNI_ERROR_EXCEPTION_THROWN);
+        tlvBytes    = reinterpret_cast<uint8_t *>(tlvBytesObjBytes);
+        hasValidTlv = true;
+    }
+    else
+    {
+        SuccessOrExit(err = JniReferences::GetInstance().FindMethod(env, invokeElement, "getJsonString", "()Ljava/lang/String;",
+                                                                    &getJsonStringMethod));
+        jstring jsonJniString = static_cast<jstring>(env->CallObjectMethod(invokeElement, getJsonStringMethod));
+        VerifyOrExit(!env->ExceptionCheck(), err = CHIP_JNI_ERROR_EXCEPTION_THROWN);
+        if (jsonJniString != nullptr)
+        {
+            JniUtfString jsonUtfJniString(env, jsonJniString);
+            uint8_t buf[chip::app::kMaxSecureSduLengthBytes] = { 0 };
```

Comment:
This buffer is valid within the scope of `{}`. tlvBytes makes it escape that, so you would be using invalid data.